flang/lib/Lower/NumericRuntime.cpp

```diff
@@ -0,0 +1,120 @@
+//===-- NumericRuntime.cpp -- runtime for numeric intrinsics -------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "../../runtime/numeric.h"
+#include "RTBuilder.h"
+#include "flang/Lower/Bridge.h"
+#include "flang/Lower/CharacterExpr.h"
+#include "flang/Lower/FIRBuilder.h"
+#include "flang/Lower/Support/BoxValue.h"
+#include "flang/Lower/Todo.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "flang/Lower/NumericRuntime.h"
+
+using namespace Fortran::runtime;
+
+// The real*10 and real*16 placeholders below are used to force the
+// compilation of the real*10 and real*16 method names on systems that
+// may not have them in their runtime library. This can occur in the
+// case of cross compilation, for example.
+
+/// Placeholder for real*10 version of RRSpacing Intrinsic
+struct ForcedRRSpacing10 {
```

Comment:
Looks a good solution to me. You might want to add one small test for one of the REAL(10) or REAL(16) version since this uses a different path than the rest.